support/Pipelines/Bbh/EccentricityControlMinimal.py

```diff
@@ -0,0 +1,647 @@
+    #         ("function", lambda p, t: p[0] * np.cos(p[1] * t + p[2]) + p[3]),
+    #         ("initial guess", [0.011, 0.014, 0, 0]), #Vittoria Tommasini,
+    #          changed initial guess for p[1], p[2]
+    #     ]
+    # )
+
+    # ==== linear + cos ====
+    functions["H3"] = dict(
+        [
+            ("label", "B*cos(w*t+phi)+linear"),
+            (
+                "function",
+                lambda p, t: p[3] + p[4] * t + p[0] * np.cos(p[1] * t + p[2]),
+            ),
+            (
+                "initial guess",
+                [
+                    0.011,
+                    0.014,  # Vittoria Tommasini , changed initial guess for
+                    # p[1], p[2]
+                    0,
+                    0,
+                    0,
+                ],
+            ),
+        ]
+    )
+
+    # ==== quadratic + cos ====
+    functions["H4"] = dict(
+        [
+            ("label", "B*cos(w*t+phi)+quadratic"),
+            (
+                "function",
+                lambda p, t: p[3]
+                + p[4] * t
+                + p[5] * t**2
+                + p[0] * np.cos(p[1] * t + p[2]),
+            ),
+            (
+                "initial guess",
+                [
+                    0.011,
+                    0.014,  # Vittoria Tommasini , changed initial guess for
+                    # p[1], p[2]
+                    0,
+                    0,
+                    0,
+                    0,
+                ],
+            ),
+        ]
+    )
+
+    # Fit and compute updates
+    for name, func in functions.items():
+        # We will handle H4 separately
+        if name == "H4":
+            continue
+
+        func["fit result"] = compute_coord_sep_updates(
+            x=t,
+            y=dsdt,
+            model=func,
+            initial_separation=initial_separation,
+            initial_xcts_values=initial_xcts_values,
+        )
+
+    # ==== quadratic + cos ====
+    # Replace the initial guess with that of the linear solve
+    iguess_len = len(functions["H3"]["initial guess"])
+    functions["H4"]["initial guess"][0:iguess_len] = functions["H3"][
+        "fit result"
+    ]["parameters"][0:iguess_len]
+
+    functions["H4"]["fit result"] = compute_coord_sep_updates(
+        x=t,
+        y=dsdt,
+        model=functions["H4"],
+        initial_separation=initial_separation,
+        initial_xcts_values=initial_xcts_values,
+    )
+
+    # Print results and plot
+    coordinate_separation_eccentricity_control_digest_H4(
+        h5_file=h5_file,
+        x=t,
+        y=dsdt,
+        data=data,
+        functions=functions,
+        output=output,
+    )
+
+    # return functions
+    # return [functions["H4"]["fit result"]['eccentricity'],  functions["H4"]
+    # ["fit result"]['xcts updates']]
+    return functions["H4"]["fit result"]
+
+
+def ecc_control_fit(
+    h5_file,
+    subfile_name_aha,
+    subfile_name_ahb,
+    tmin,
+    tmax,
+    angular_velocity_from_xcts,
```

Comment:
Instead of these ID params, pass in the path to the initial data input file (name it `id_input_file_path`). Then read the orbital params out of the YAML file (see `support/Pipelines/Bbh/ControlId.py:120`).